### ISSUE TITLE
Printer casts native-int constant to wrong signedness → CS0034 (nuint op nint) at Full

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#1459]
## Goal

Over-raise correctness bug for the #1356 burndown, surfaced by a `--validity-check` sweep of `System.Private.CoreLib`. The printer casts a native-int constant to the **wrong signedness** in a binary/compound op, so an operation between `nuint` and `nint` binds ambiguously (CS0034) while the method is reported `Full` (15 diagnostics in the sweep).

## The over-raise claim being narrowed

C# requires the two operands of an arithmetic op to be the same native-int signedness. When the decompiler renders a constant operand of a native-int op, it casts it to `nint` regardless of the other operand's type. Against a `nuint` operand that yields `nuint op nint`, which C# rejects as ambiguous (`CS0034: Operator '-=' is ambiguous on operands of type 'nuint' and 'nint'`).

## Minimal counterexample (real CoreLib method, reproduced through the pipeline)

`System.Buffer::BulkMoveWithWriteBarrierBatch` decompiles to (at `Fidelity=Full`):

```csharp
byteCount -= ((nint)16384);     // byteCount is nuint → CS0034 (nuint -= nint, ambiguous)
```

`byteCount` is `nuint`; the constant must be `(nuint)16384`. Reproduced via the real pipeline.

## Missing gate

The native-int constant cast does not take the sibling operand's signedness into account; it always emits `(nint)`.

## First fix shape

When emitting a native-int constant (or any cast-to-native-int) operand of a binary/compound op, choose the cast signedness from the operation's other operand / result type — `(nuint)` when the partner is `nuint`, `(nint)` otherwise. This mirrors the unsigned-operand coercion the printer already applies for fixed-width comparisons (`(uint)a < (uint)b`).

## Done signal

- `Buffer::BulkMoveWithWriteBarrierBatch` (and the other CS0034 native-int cases) recompile (matching `nuint`/`nint` cast).
- `nint op nint` and fixed-width int ops are unchanged.

## Notes

Found by the `--validity-check` CoreLib sweep (CS0034 ×15). Sign-handling sibling of #1452 (operand-coercion gap) and the #1425/#1403 conversion family. Distinct root from the CS0266 pointer-cast drop filed separately.

**Comment by richlander** [comment]
Claiming as burndown runner from #1598. Narrow slice: when emitting a native-int constant/cast operand of a binary or compound op, choose `(nuint)` vs `(nint)` from the sibling/result signedness (mirroring the existing fixed-width unsigned-operand coercion). Will add a focused fixture for the `nuint -= (nuint)const` shape plus `nint`/fixed-width positive canaries; no printer widening.

**Comment by richlander** [comment]
PR #1619 opened. Scope note: the issue's measured cases (the constant `-=`/native-int sweep, CS0034 ×15) are already resolved on current main by the mixed-sign arithmetic work — a fresh CoreLib `--validity-check` sweep reports **0 CS0034**, and `Buffer::BulkMoveWithWriteBarrierBatch` recompiles as `byteCount -= (nuint)((nint)16384)`. PR #1619 closes the two remaining native-int compound holes found while verifying: sign-sensitive `/=`/`%=` (excluded from reconciliation) and the `ref nuint` indirect form (lvalue signedness lost through `ldind.i`). Defect set is byte-identical before/after (zero corpus delta).
